server/app.js

```diff
@@ -27,9 +32,37 @@ const io = socketio(server, {
     origin: "*",
   },
 });
+io.use(ioCookieParser());
+io.use((socket, next) => {
+  const token = socket.request.cookies["token"];
+  if (token) {
+    try {
+      const decoded = jwt.verify(token, process.env.JWT_SECRET);
+      user = decoded;
+      next();
+    } catch (err) {
+      return next(new Error("Authentication Error"));
+    }
+  } else {
+    return next(new Error("Authentication Error"));
+  }
+}).on("connection", (socket, next) => {
+  socket.on("addUser", (userId) => {
+    const users = addUser(userId, socket.id);
+    io.emit("getUsers", users);
+  });
+
+  socket.on("sendMessage", ({ recipientId, ...rest }) => {
+    const activeUser = getUser(recipientId);
+    if (activeUser) {
+      io.to(activeUser.socketId).emit("getMessage", { ...rest });
+    }
+  });
 
-io.on("connection", (socket) => {
-  console.log("connected");
+  socket.on("disconnect", () => {
+    const users = removeUser(socket.id);
```

Comment:
I think one user at a time. So you may use `user` rather.

Reply:
done ✔️